src/runtime/graph/graph_runtime.cc

```diff
@@ -105,25 +106,78 @@ class GraphRuntime : public ModuleNode {
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
     TVM_CCALL(TVMArrayCopyFromTo(data_in, &data_entry_[eid], nullptr));
   }
+  /*!
+   * \brief Get the input count
+   *
+   * \return The count of inputs
+   */
+  int GetInputCount(void) {
+    return input_nodes_.size();
+  }
+  /*!
+   * \brief Get the outputs count
+   *
+   * \return The count of outputs
+   */
+  int GetOutputCount(void) {
+    return outputs_.size();
+  }
+  /*!
+   * \brief Return NDArray for given input index.
+   * \param index The input index.
+   *
+   * \return NDArray corresponding to given input node index.
+   */
+  NDArray GetInputAsNDArray(int index) {
+    CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
+    uint32_t eid = this->entry_id(input_nodes_[index], 0);
+    return NDArray::FromDLTensor(&data_entry_[eid]);
+  }
+  /*!
+   * \brief Return NDArray for given output index.
+   * \param index The output index.
+   *
+   * \return NDArray corresponding to given output node index.
+   */
+  NDArray GetOutputAsNDArray(int index) {
+    CHECK_LT(static_cast<size_t>(index), outputs_.size());
+    uint32_t eid = this->entry_id(outputs_[index]);
+    tvm::runtime::DeviceAPI::Get(GetCtx())->StreamSync(GetCtx(), nullptr);
```

Comment:
Is this sync really necessary here?

Reply:
Sync here is to make sure the GPU operations completed before taking output.